scripts/lib/CIME/nmlgen.py

```diff
@@ -274,12 +274,24 @@ def get_default(self, name, config=None, allow_none=False):
                     default[i] = self.quote_string(scalar)
 
         default = self._to_python_value(name, default)
+
         return default
 
     def get_streams(self):
         """Get a list of all streams used for the current data model  mode."""
         return self.get_default("streamslist")
 
+    def clean_streams(self):
+        for variable in self._streams_variables:
+            self._streams_namelists[variable] = []
+        self._streams_namelists["streams"] = []
+
+    def new_instance(self):
```

Comment:
Clean data stored from previous instances so that it isn't copied into new instance by instance here I am refering to the cesm ensemble mode functionality.